File: server/Models/Vault.cs
using System.ComponentModel.DataAnnotations;
using keepr.interfaces;

namespace keepr.Models;

// INPUT MODEL
public class VaultCreationDTO
{
  public string Name { get; set; }
  [MaxLength(1000)]
  public string Description { get; set; }
  [MaxLength(1000)]
  public string Img { get; set; }
}

// RETURN OBJECT
public class Vault : IRepoItem<int>
{
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public string Img { get; set; }
  public bool IsPrivate { get; set; }

  // RELATIONSHIP PROPERTIES
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: client/src/models/Vault.js
import { DBItem } from "./DBItem.js";
import { Profile } from "./Profile.js";

export class Vault extends DBItem{
  constructor(data){
    super(data)
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.isPrivate = data.isPrivate
    this.creatorId = data.creatorId
    this.creator = data.creator ? new Profile(data.creator) : null
  }
}